warden/app/legacy.go

```diff
@@ -260,3 +265,9 @@ func RegisterLegacyModules(registry cdctypes.InterfaceRegistry) map[string]appmo
 
 	return modules
 }
+
+func WardenProtocolCustomEncoder() wasmkeeper.MessageEncoders {
+	return wasmkeeper.MessageEncoders{
```

Comment:
I'm wondering if this overrides completely the default MessageEncoders struct, and maybe we should construct it using [DefaultEncoders()](https://pkg.go.dev/github.com/CosmWasm/wasmd/x/wasm/keeper#DefaultEncoders) and setting the `Custom` field there, but I haven't checked.

Reply:
No, it doesn't override default encoders completely. It just replaces encoder for the custom message (which is empty by default).

Here's [the code](https://github.com/CosmWasm/wasmd/blob/v0.50.0/x/wasm/keeper/handler_plugin_encoders.go#L60) that does this.

Some additional details:
- constructor of the wasm keeper [creates](https://github.com/CosmWasm/wasmd/blob/v0.50.0/x/wasm/keeper/keeper_cgo.go#L50) default message handler.
- which in turn [initializes](https://github.com/CosmWasm/wasmd/blob/main/x/wasm/keeper/handler_plugin.go#L51) default encoders. Note that default custom encoder is [empty](https://github.com/CosmWasm/wasmd/blob/main/x/wasm/keeper/handler_plugin_encoders.go#L50).
- after the basic initialization of the wasm keeper is done, its constructor [applies](https://github.com/CosmWasm/wasmd/blob/v0.50.0/x/wasm/keeper/keeper_cgo.go#L64) custom wasm options.
- you can alter the default encoder by [passing](https://github.com/warden-protocol/wardenprotocol/blob/d07e0bb8c97c18d3fd43fb5cd4bf4d2758454ac5/warden/app/legacy.go#L169-L170) corresponding option. Note that implementation of `WithMessageEncoders` [changes](https://github.com/CosmWasm/wasmd/blob/main/x/wasm/keeper/handler_plugin.go#L53) only those parts of the encoder that [were requestsed](https://github.com/CosmWasm/wasmd/blob/v0.50.0/x/wasm/keeper/handler_plugin_encoders.go#L60).